docs/training/mixed-precision.md

```diff
@@ -0,0 +1,165 @@
+# Mixed Precision Training
```

Comment:
doesnt have to be in this pr, but we need to explain when mixed precision is on, which components are in fp8 / bf16 / fp32. maybe tag someone from @erhoo82 's team?